benchmarks/renderImageItem.py

```diff
@@ -22,117 +23,183 @@
-
-    @staticmethod
-    def _create_data(size, xp):
-        float_data = {
-            "data": xp.random.normal(size=size).astype("float32"),
-            "levels": [-4.0, 4.0],
-        }
-        uint16_data = {
-            "data": xp.random.randint(100, 4500, size=size).astype("uint16"),
-            "levels": [250, 3000],
-        }
-        uint8_data = {
-            "data": xp.random.randint(0, 255, size=size).astype("ubyte"),
-            "levels": [20, 220],
-        }
-        c_map = xp.array([[-500.0, 255.0], [-255.0, 255.0], [0.0, 500.0]])
-        uint8_lut = xp.zeros((256, 4), dtype="ubyte")
-        for i in range(3):
-            uint8_lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 256), 0, 255)
-        uint8_lut[:, 3] = 255
-        uint16_lut = xp.zeros((2 ** 16, 4), dtype="ubyte")
-        for i in range(3):
-            uint16_lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 2 ** 16), 0, 255)
-        uint16_lut[:, 3] = 255
-        return float_data, uint16_data, uint8_data, uint16_lut, uint8_lut
-
-
-def make_test(dtype, kind, use_levels, lut_name, func_name):
-    def time_test(self):
-        data = getattr(self, dtype + "_data")
-        levels = data["levels"] if use_levels else None
-        lut = getattr(self, f"{kind}_{lut_name}_lut", None) if lut_name is not None else None
-        pg.setConfigOption("useNumba", kind == "numba")
-        img_data = data["data"]
-        if kind == "cupy":
-            img_data = cp.asarray(img_data)
-        renderQImage(img_data, lut=lut, levels=levels)
-
-    time_test.__name__ = func_name
-    return time_test
-
-
-for option in ["cupy", "numba", "numpy"]:
-    if option == "cupy" and cp is None:
-        continue
-    if option == "numba" and numba is None:
-        continue
-    for data_type in ["float", "uint16", "uint8"]:
-        for lvls in [True, False]:
-            if data_type == "float" and not lvls:
-                continue
-            for lutname in [None, "uint8", "uint16"]:
-                name = (
-                    f'time_1x_renderImageItem_{option}_{data_type}_{"" if lvls else "no"}levels_{lutname or "no"}lut'
-                )
-                setattr(_TimeSuite, name, make_test(data_type, option, lvls, lutname, name))
-
-
-class Time4096Suite(_TimeSuite):
-    def __init__(self):
-        super(Time4096Suite, self).__init__()
-        self.size = 4096
+
+        if acceleration == "cupy":
+            if cp is None:
+                raise NotImplementedError
+            pg.setConfigOption("useCupy", True)
+            xp = cp  # use cupy instead of numpy
+
+        # these codepaths bypass makeARRGB
+        if acceleration != "numpy":
+            if xp.dtype(dtype) == xp.ubyte and not use_levels:
+                if lut_length is None:
+                    # Grayscale8, RGB888 or RGB[AX]8888
+                    raise NotImplementedError
+                if channels == 1 and xp.dtype(lut_length) == xp.uint8:
```

Comment:
While this is technically correct; I had to scroll back a ways and think through it a fair amount before I agreed with the static code checker; I'm thinking this might be a case where I should disregard it and keep the check as-is.